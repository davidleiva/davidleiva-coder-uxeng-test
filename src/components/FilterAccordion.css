.MuiAccordion-root.MuiPaper-root {
    background-color: rgb(4, 6, 11);
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    box-shadow: none;
    background-image: none;
    color: hsl(218, 32%, 77%);
  }
  
  .MuiAccordion-root.MuiPaper-root svg path {
    fill: hsl(218, 32%, 77%);
  }
  
  .MuiAccordion-root.MuiPaper-root:hover,
  .MuiAccordion-root.MuiPaper-root.Mui-expanded {
    color: hsl(220, 55%, 98%);
  }
  
  .MuiAccordion-root.MuiPaper-root:hover svg path {
    color: hsl(220, 55%, 98%);
  }
  
  .MuiAccordionDetails-root {
    padding: 0;
  }
  
  .MuiAccordion-root.MuiPaper-root .MuiAccordionSummary-root {
    padding-left: 0;
    padding-right: 0;
    height: 36px;
  }
  
  .MuiAccordionSummary-content > .MuiTypography-root,
  .MuiListItemText-root > .MuiTypography-root,
  .FilterSideBar .MuiAccordionSummary-content > .MuiTypography-root,
  .MuiAccordionSummary-content > .MuiTypography-root,
  .FilterSideBar .MuiListItemText-root > .MuiTypography-root {
    font-size: 12px;
  }
  
  .MuiAccordion-root.MuiPaper-root:before {
    content: none;
  }
  
  .MuiAccordion-root.MuiPaper-root .MuiSvgIcon-root {
    width: 0.66em;
    height: 0.66em;
  }
  
  .MuiAccordionSummary-content.Mui-expanded {
    margin: 4px 0;
  }
  
  .MuiAccordion-root .MuiButtonBase-root {
    min-height: initial;
    height: 36px;
  }
  
  .MuiAccordion-root.MuiPaper-root.Mui-expanded {
    margin-bottom: 0;
    margin-top: 0;
  }
  
  .MuiListItemButton-root.justifyEnd .MuiListItemIcon-root {
    justify-content: end;
  }

  .FilterBar__Divider.MuiDivider-root {
    margin: 8px auto;
  }

.FilterSearchUser input, .FilterSearchUser label,
.FilterSearchUser .MuiAvatar-circular {
  font-size: 12px;
}

.MuiListItemButton-root .MuiListItemIcon-root, .MuiListItemButton-root .MuiAvatar-root {
  min-width: 0;
}

.MuiListItemButton-root .MuiAvatar-root {
  margin-right: 8px;
}