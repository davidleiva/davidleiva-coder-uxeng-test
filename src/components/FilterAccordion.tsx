import Accordion from '@mui/material/Accordion';
import AccordionSummary from '@mui/material/AccordionSummary';
import AccordionDetails from '@mui/material/AccordionDetails';
import Typography from '@mui/material/Typography';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { FiltersSB } from './FiltersSB';

export interface FilterAccordionProps {
    title: string,
    defaultExp?: boolean,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    filters: any,
    isExpanded: boolean,
    // eslint-disable-next-line @typescript-eslint/ban-types
    handleChange: Function,
    filterState?: object,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    type: any
}

export const FilterAccordion = ({
    title,
    defaultExp,
    filters,
    isExpanded,
    handleChange,
    type
    }: FilterAccordionProps) => {
    
    return (
        <Accordion
            expanded={isExpanded}
            sx={{
                bgcolor: 'transparent',
            }}
            defaultExpanded={defaultExp}
            onChange={() => handleChange()}
        >
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
                aria-controls="panel1a-content"
                id="panel1a-header"
            >
                <Typography>{title}</Typography>
            </AccordionSummary>
            <AccordionDetails>
                <FiltersSB filters={filters} type={type}/>
            </AccordionDetails>
        </Accordion>
      )
};

export default FilterAccordion;