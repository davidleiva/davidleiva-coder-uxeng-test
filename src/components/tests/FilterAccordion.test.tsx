import {describe, expect, test} from 'vitest';
import FilterAccordion from '../FilterAccordion';

describe("Accordion test", () => {
    test("FilterAccordion renders ok", () => {
        
        const fc = 
            <FilterAccordion 
                title={'title'}
                defaultExp={true}
                filters={[]}
                isExpanded={true}
                handleChange={() => {}}
                filterState={{}}
                type={''}
            />
        ;

        expect(fc).toMatchSnapshot()
    })
})