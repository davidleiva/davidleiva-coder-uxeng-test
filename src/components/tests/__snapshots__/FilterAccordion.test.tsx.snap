// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion test > FilterAccordion renders ok 1`] = `
<FilterAccordion
  defaultExp={true}
  filterState={{}}
  filters={[]}
  handleChange={[Function]}
  isExpanded={true}
  title="title"
  type=""
/>
`;
